### ISSUE TITLE
DeprecationWarning: HTTPResponse.getheader() is deprecated

### ISSUE THREAD

**Issue by alexerhardt** [pinecone-io/pinecone-python-client#125]
I constantly get the following error message:

`/path-to-environment/lib/python3.8/site-packages/pinecone/core/client/rest.py:45: DeprecationWarning: HTTPResponse.getheader() is deprecated and will be removed in urllib3 v2.1.0. Instead use HTTResponse.headers.get(name, default).`

I can see in the source code the following:

```python
class RESTResponse(io.IOBase):
    # ...
    def getheader(self, name, default=None):
        """Returns a given response header."""
        return self.urllib3_response.getheader(name, default)
```

The warnings are very frequent, and the fix at first glance seems rather easy.

**Lib Version**
2.1.0

**OS/Hardware**
Mac OS Monterey 12.0.1 / Apple M1 Macbook Pro

**Comment by gdj0nes** [comment]
Thanks for calling this to our attention! We'll get this added to an upcoming release.

**Comment by liorgreenbergmh** [comment]
This issue still happens in 2.2.2

**Comment by daverigby** [comment]
This was fixed in v2.2.3 via: https://github.com/pinecone-io/pinecone-python-client/commit/f6bda12e59a7a034f11b5b487a957c4effaddd6a
